pkg/controller/account/account_controller.go

```diff
@@ -660,35 +672,53 @@ func CreateAccount(client awsclient.Client, accountName, accountEmail string) (*
 }
 
 // CreateIAMUser takes a client and string and creates a IAMuser
-func CreateIAMUser(client awsclient.Client, userName string) (*iam.CreateUserOutput, error) {
+func CreateIAMUser(reqLogger logr.Logger, client awsclient.Client, userName string) (*iam.CreateUserOutput, error) {
 
 	// check if username exists for this account
 	_, err := client.GetUser(&iam.GetUserInput{
 		UserName: aws.String(userName),
 	})
 
-	awserr, ok := err.(awserr.Error)
-
-	if err != nil && awserr.Code() != iam.ErrCodeNoSuchEntityException {
-		return &iam.CreateUserOutput{}, err
-	}
-
 	var createUserOutput *iam.CreateUserOutput
-	if ok && awserr.Code() == iam.ErrCodeNoSuchEntityException {
-		createResult, err := client.CreateUser(&iam.CreateUserInput{
-			UserName: aws.String(userName),
-		})
-		if err != nil {
-			return &iam.CreateUserOutput{}, err
+	// handle errors
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case iam.ErrCodeNoSuchEntityException:
+				createResult, err := client.CreateUser(&iam.CreateUserInput{
+					UserName: aws.String(userName),
+				})
+				if err != nil {
+					// Log new AWS error
+					reqLogger.Error(aerr,
+						fmt.Sprintf(`New AWS Error during creation of IAM user (ErrCodeNoSuchEntityException), 
+							AWS Error Code: %s, 
+							AWS Error Message: %s`,
+							aerr.Code(),
+							aerr.Message()))
+					return &iam.CreateUserOutput{}, err
+				}
+				createUserOutput = createResult
+			default:
+				// Log new AWS error
+				reqLogger.Error(aerr,
+					fmt.Sprintf(`New AWS Error during creation of IAM user (not ErrCodeNoSuchEntityException), 
+						AWS Error Code: %s, 
+						AWS Error Message: %s`,
+						aerr.Code(),
+						aerr.Message()))
+				return &iam.CreateUserOutput{}, err
+			}
+		} else {
```

Comment:
you don't need the else it can be omitted and  the behavior would be the same

Reply:
+1 fixed